### PR TITLE
Added new Me view, using a static tableview.

### PR DESCRIPTION
**Pull request by Elland**
screen shot 2017-05-03 at 14 37 22

### CODE REVIEW

File: Token/Controllers/Settings/SettingsController.swift [hunk 1]
```diff
@@ -86,23 +98,20 @@ open class SettingsController: SweetTableController {
...
-        self.tableView.delegate = self
-        self.tableView.separatorStyle = .none
-        self.tableView.contentInset = UIEdgeInsets(top: 20, left: 0, bottom: 0, right: 0)
+    static func instantiateFromNib() -> SettingsController {
+        return UIStoryboard(name: "Settings", bundle: nil).instantiateInitialViewController() as! SettingsController
+    }
 
-        for type in self.cellTypes {
-            self.tableView.register(type)
+    func updateVerificationStatus(_ notification: Notification) {
```
Comment (yuliaveres) [inline Token/Controllers/Settings/SettingsController.swift]: private ?

File: Token/Models/EthereumConverter.swift [hunk 2]
```diff
@@ -61,11 +61,26 @@ struct EthereumConverter {
         return "\(self.fiatValueString(forWei: balance)) \(Locale(identifier: self.forcedLocale).currencyCode!)"
     }
 
+    public static func balanceSparseAttributedString(forWei balance: NSDecimalNumber, width: CGFloat) -> NSAttributedString {
+        let attributedString: NSMutableAttributedString = self.balanceAttributedString(forWei: balance).mutableCopy() as! NSMutableAttributedString
+        let range = NSRange(location: 0, length: attributedString.length)
+
+        let paragraph = NSMutableParagraphStyle()
+        paragraph.alignment = .left
+        let t = NSTextTab(textAlignment: .right, location: width, options: [:])
```
Comment (yuliaveres) [inline Token/Models/EthereumConverter.swift]: really, let's give it better name ? 🙏

File: Token/Models/EthereumConverter.swift [hunk 3]
```diff
@@ -61,11 +61,26 @@ struct EthereumConverter {
...
+        paragraph.tabStops = [t]
+
+        attributedString.addAttribute(NSParagraphStyleAttributeName, value: paragraph, range: range)
+        print(attributedString)
```
Comment (yuliaveres) [inline Token/Models/EthereumConverter.swift]: needed ?

File: Token/Models/EthereumConverter.swift [hunk 4]
```diff
@@ -61,11 +61,26 @@ struct EthereumConverter {
         return "\(self.fiatValueString(forWei: balance)) \(Locale(identifier: self.forcedLocale).currencyCode!)"
     }
 
+    public static func balanceSparseAttributedString(forWei balance: NSDecimalNumber, width: CGFloat) -> NSAttributedString {
```
Comment (yuliaveres) [inline Token/Models/EthereumConverter.swift]: let's stick to the following:
https://swift.org/documentation/api-design-guidelines/
then I would write `public static func balanceSparseAttributedString(for wei: NSDecimalNumber, width: CGFloat) -> NSAttributedString {`

File: Token/Controllers/Settings/SettingsController.swift [hunk 5]
```diff
@@ -38,42 +38,54 @@ open class SettingsController: SweetTableController {
...
-        NotificationCenter.default.addObserver(self, selector: #selector(self.updateVerificationStatus(_:)), name: SettingsController.verificationStatusChanged, object: nil)
+    public var chatAPIClient: ChatAPIClient {
+        return ChatAPIClient.shared
     }
 
-    func updateVerificationStatus(_ notification: Notification) {
-        if let verificationStatus = notification.object as? VerificationStatus {
-            self.verificationStatus = verificationStatus
-        }
+    public var idAPIClient: IDAPIClient {
```
Comment (yuliaveres) [inline Token/Controllers/Settings/SettingsController.swift]: do we need it public ? in other controllers it is not, and seems like not used outside :/ 
please check, maybe possible private ?

File: Token/Controllers/Settings/SettingsController.swift [hunk 6]
```diff
@@ -148,80 +157,72 @@ open class SettingsController: SweetTableController {
...
 
-                    if self.didVerifyBackupPhrase == true, cell.checkbox.checked == false {
-                        cell.checkbox.bounce()
-                    }
+        case 2:
+            switch indexPath.row {
+            case 0:
+                // passphrase backup
+                self.navigationController?.pushViewController(BackupPhraseEnableController(), animated: true)
+            case 1:
```
Comment (yuliaveres) [inline Token/Controllers/Settings/SettingsController.swift]: why we need this case if default breaks?

File: Token/Controllers/Settings/SettingsController.swift [hunk 7]
```diff
@@ -148,80 +157,72 @@ open class SettingsController: SweetTableController {
...
+                // go to user profile
+                self.navigationController?.pushViewController(ProfileController(), animated: true)
+            case 1:
+                break // go to qr code
+            default:
+                break
+            }
+        case 1:
+            switch indexPath.row {
+            case 1:
```
Comment (yuliaveres) [inline Token/Controllers/Settings/SettingsController.swift]: same as in second section

File: Token/Controllers/Settings/SettingsController.swift [hunk 8]
```diff
@@ -148,80 +157,72 @@ open class SettingsController: SweetTableController {
...
+                break // trusted frieds
+            default:
+                break
+            }
 
-                    cell.checkbox.checked = self.didVerifyBackupPhrase
-                }
+        case 3:
+            switch indexPath.row {
+            case 0:
```
Comment (yuliaveres) [inline Token/Controllers/Settings/SettingsController.swift]: and here

File: Token/Controllers/Settings/SettingsController.swift [hunk 9]
```diff
@@ -148,80 +157,72 @@ open class SettingsController: SweetTableController {
...
+    open override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        self.performAction(for: indexPath)
     }
 
-    public func tableView(_: UITableView, estimatedHeightForRowAt _: IndexPath) -> CGFloat {
-        return UITableViewAutomaticDimension
-    }
+    open override func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        if section == 2 {
+            let view = SettingsSectionHeader(title: "Security", error: "Your account is at risk")
```
Comment (yuliaveres) [inline Token/Controllers/Settings/SettingsController.swift]: i think if we target for locales in future, lets already use NSLocalized.. while adding new code ?

**Comment by yuliaveres** [comment]
Also, doesn't it look like only "My QR Code" gets aligned with sections titles, but rest cells not ? ;(

File: Token/Controllers/Settings/SettingsController.swift (hunk 1 above)
Comment (Elland) [inline Token/Controllers/Settings/SettingsController.swift]: If they're private, ObjectiveC runtime can't see it. This is a notification handler, gotta be internal at least.

File: Token/Controllers/Settings/SettingsController.swift (hunk 6 above)
Comment (Elland) [inline Token/Controllers/Settings/SettingsController.swift]: These are placeholders for the actions we'll implement later. I like having this already scaffolded here, so I don't have to double check the indices again later. 👍

File: Token/Controllers/Settings/SettingsController.swift (hunk 9 above)
Comment (Elland) [inline Token/Controllers/Settings/SettingsController.swift]: There was a previous discussion about how to improve on the localisation thing. We can discuss it tomorrow all in person!

File: Token/Models/EthereumConverter.swift (hunk 4 above)
Comment (Elland) [inline Token/Models/EthereumConverter.swift]: Then we can't differentiate between `for wei` and `for ether`. They'll conflict.

File: Token/Models/EthereumConverter.swift (hunk 2 above)
Comment (Elland) [inline Token/Models/EthereumConverter.swift]: Hoopla, forgot to rename this. Thanks. :)

File: Token/Controllers/Settings/SettingsController.swift (hunk 9 above)
Comment (roberthein) [inline Token/Controllers/Settings/SettingsController.swift]: Ok. 👌 But I think in general it's never a good idea to use hardcoded strings.

Comment (Elland) [inline Token/Controllers/Settings/SettingsController.swift]: Agreed!

**Comment by Elland** [comment]
Approved? 🤖

File: Token/Controllers/Settings/SettingsController.swift (hunk 1 above)
Comment (yuliaveres) [inline Token/Controllers/Settings/SettingsController.swift]: u can mark @objc private func

File: Token/Models/EthereumConverter.swift (hunk 4 above)
Comment (yuliaveres) [inline Token/Models/EthereumConverter.swift]: sad :(

**Comment by Elland** [comment]
Alright. All set?

**Pull request merged**